execution/block-partitioner/src/sharded_block_partitioner/messages.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    sharded_block_partitioner::dependency_analysis::{RWSet, RWSetWithTxnIndex},
+    types::{SubBlock, TxnIndex},
+};
+use aptos_types::transaction::analyzed_transaction::AnalyzedTransaction;
+use std::sync::Arc;
+
+pub enum ControlMsg {
+    DiscardCrossShardDepReq(DiscardTxnsWithCrossShardDep),
+    AddCrossShardDepReq(AddTxnsWithCrossShardDep),
+    Stop,
+}
+
+#[derive(Clone, Debug)]
+pub enum CrossShardMsg {
+    RWSetWithTxnIndexMsg(RWSetWithTxnIndex),
+    RWSetMsg(RWSet),
+    // Number of accepted transactions in the shard for the current round.
+    AcceptedTxnsMsg(usize),
+}
+
+pub struct DiscardTxnsWithCrossShardDep {
+    pub transactions: Vec<AnalyzedTransaction>,
+    // The frozen dependencies in previous chunks.
+    pub prev_rounds_rw_set_with_index: Arc<Vec<RWSetWithTxnIndex>>,
+    pub prev_rounds_frozen_chunks: Arc<Vec<SubBlock>>,
+}
+
+impl DiscardTxnsWithCrossShardDep {
+    pub fn new(
+        transactions: Vec<AnalyzedTransaction>,
+        prev_rounds_rw_set_with_index: Arc<Vec<RWSetWithTxnIndex>>,
+        prev_rounds_frozen_chunks: Arc<Vec<SubBlock>>,
+    ) -> Self {
+        Self {
+            transactions,
+            prev_rounds_rw_set_with_index,
+            prev_rounds_frozen_chunks,
+        }
+    }
+}
+
+pub struct AddTxnsWithCrossShardDep {
+    pub transactions: Vec<AnalyzedTransaction>,
+    pub index_offset: TxnIndex,
+    pub prev_rounds_frozen_chunks: Arc<Vec<SubBlock>>,
+    // The frozen dependencies in previous chunks.
+    pub prev_rounds_rw_set_with_index: Arc<Vec<RWSetWithTxnIndex>>,
+}
+
+impl AddTxnsWithCrossShardDep {
+    pub fn new(
+        transactions: Vec<AnalyzedTransaction>,
+        index_offset: TxnIndex,
+        prev_rounds_frozen_chunks: Arc<Vec<SubBlock>>,
+        prev_rounds_rw_set_with_index: Arc<Vec<RWSetWithTxnIndex>>,
+    ) -> Self {
+        Self {
+            transactions,
+            index_offset,
+            prev_rounds_rw_set_with_index,
+            prev_rounds_frozen_chunks,
+        }
+    }
+}
+
+pub struct PartitioningBlockResponse {
+    pub frozen_chunk: SubBlock,
+    pub rw_set_with_index: RWSetWithTxnIndex,
+    pub rejected_txns: Vec<AnalyzedTransaction>,
```

Comment:
"rejected" and "discarded" seem to mean the same thing across the change, how about unifying the concept?

Reply:
Renamed all "rejected" -> "discarded".